### PR TITLE
Add support for api key metadata

### PR DESCRIPTION
**Pull request by victor88121**
https://linear.app/nightfall/issue/PLAT-2278/asd-nightfall-code-scanner
Screen Shot 2022-07-29 at 6 57 33 PM

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -115,8 +115,8 @@ github.com/kr/pretty v0.1.0/go.mod h1:dAy3ld7l9f0ibDNOQOHHMYYIIbhfbHSm3C4ZsoJORN
 github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
 github.com/kr/text v0.1.0 h1:45sCR5RtlFHMR4UwH9sdQ5TC8v0qDQCHnXt+kaKSTVE=
 github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
-github.com/nightfallai/nightfall-go-sdk v1.1.1 h1:xx/e4ZP415oHQcJMOCBeU8kxOWiakrjtEEREZHJtE8Q=
-github.com/nightfallai/nightfall-go-sdk v1.1.1/go.mod h1:IQxS8JqhFEGRfnOpaLqv+ZEhQrUKbJgFAQ5qHU3Lcbg=
+github.com/nightfallai/nightfall-go-sdk v1.1.5 h1:QJ6SxZKTqmUDK/Jh2m4I3Dhk1NMJ8ThPX4aZVfnugwA=
+github.com/nightfallai/nightfall-go-sdk v1.1.5/go.mod h1:IQxS8JqhFEGRfnOpaLqv+ZEhQrUKbJgFAQ5qHU3Lcbg=
 github.com/pkg/errors v0.9.1 h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=
```
Comment (visaals) [inline go.sum:120]: @evanfuller you're right that we need to be careful detecting random keys in `go.sum` files 

cc @martinmr

**Review by visaals — APPROVED** [review]
LGTM

File: go.sum (hunk 1 above)
Comment (evanfuller) [reply go.sum:120]: oh also @martinmr not sure where the string is defined by the capitalization looks wrong for Git_**L**_ab

Comment (evanfuller) [reply go.sum:120]: agh, crap, never mind this is a function of the logic below. @victor88121 can we remove the casing logic for the key kinds? we can assume that they've been properly set in the messages, e.g. we'll send `GitHub` or `GitLab` down the wire, we don't want to mistakenly update to `Github` or `Gitlab`

**Pull request merged**